lib/contract.js

```diff
@@ -180,24 +188,40 @@ var addEventsToContract = function (contract, desc, address) {
  *      outputs: [{name: 'd', type: 'string' }]
  * }];  // contract abi
  *
- * var myContract = web3.eth.contract('0x0123123121', abi); // creation of contract object
+ * var MyContract = web3.eth.contract(abi); // creation of contract prototype
+ *
+ * var contractInstance = new MyContract('0x0123123121');
  *
- * myContract.myMethod('this is test string param for call'); // myMethod call (implicit, default)
- * myContract.call().myMethod('this is test string param for call'); // myMethod call (explicit)
- * myContract.transact().myMethod('this is test string param for transact'); // myMethod transact
+ * contractInstance.myMethod('this is test string param for call'); // myMethod call (implicit, default)
+ * contractInstance.call().myMethod('this is test string param for call'); // myMethod call (explicit)
+ * contractInstance.sendTransaction().myMethod('this is test string param for transact'); // myMethod sendTransaction
  *
- * @param address - address of the contract, which should be called
- * @param desc - abi json description of the contract, which is being created
+ * @param abi - abi json description of the contract, which is being created
  * @returns contract object
  */
+var contract = function (abi) {
 
-var contract = function (address, desc) {
+    // return prototype
+    if(Object.prototype.toString.call(abi) === '[object Array]' && arguments.length === 1) {
```

Comment:
can you make utils methods for checking types? eg.

```javascript
var isFunction = function (..) {}
var isArray = functin (...) {}
// and so on
```

also i would prefer to use `x instanceof Array`. Or you should check how underscore or lodash is checking if something is array.